src/identity/mod.rs

```diff
@@ -37,9 +33,45 @@ impl PublicId {
     /// Returns the entity's network address.
     pub fn name(&self) -> &XorName {
         match self {
-            PublicId::Node(pub_id) => pub_id.name(),
-            PublicId::Client(pub_id) => pub_id.name(),
-            PublicId::App(pub_id) => pub_id.owner_name(),
+            Self::Node(pub_id) => pub_id.name(),
+            Self::Client(pub_id) => pub_id.name(),
+            Self::App(pub_id) => pub_id.owner_name(),
+        }
+    }
+
+    /// Returns the node public id, if applicable.
+    pub fn node_public_id(&self) -> Option<&node::PublicId> {
+        if let Self::Node(id) = self {
+            Some(id)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the client public id, if applicable.
+    pub fn client_public_id(&self) -> Option<&client::PublicId> {
+        if let Self::Client(id) = self {
+            Some(id)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the app public id, if applicable.
+    pub fn app_public_id(&self) -> Option<&app::PublicId> {
+        if let Self::App(id) = self {
+            Some(id)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the entity's public key, if applicable.
+    pub fn public_key(&self) -> Option<PublicKey> {
+        match self {
+            Self::Node(pub_id) => pub_id.bls_public_key().map(|key| key.into()),
```

Comment:
Do we specifically need the BLS public key?

Reply:
What should be returned here? The ed25519 key?

Reply:
Yes I believe so. The vault by default doesn't have a BLS key, it generates only the ed25519 key and that's the key that represents the Vault's identity.